### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by yunfeiguo**
- [x] [RES-423](https://binatechnologies.atlassian.net/browse/RES-423)
- [x] Blocking PRs: ADD HERE IF ANY
* People
  - [x] Reviewers (>=1): @bayolau 
  - [x] Merger (>=1): @bayolau 
  - [x] FYI: @marghoob 
- [x] Release note (if needed)
- [x] No code copied from outside Bina 
- [x] Tests (Unit, Workflow, ITL & Gherkin) are passing
- [x] PR labels, milestones & assignees

# Summary

* refactor some public classes: more accurate representation
* refactor `EnumDat` for more robust handling of pbbam tags
* fix some typos
* converted file types and suffixes into enum

### CODE REVIEW

File: src/main/java/com/bina/lrsim/SamplerDriver.java [hunk 1]
```diff
@@ -76,17 +76,23 @@ else if(this.reference.exists()) {
         }
         return null;
       } else if (readType.equals("bax")) {
-        if (this.inFile.endsWith("cmp.h5")) {
+        if (this.inFile.endsWith(".cmp.h5")) {
```
Comment (marghoob) [inline src/main/java/com/bina/lrsim/SamplerDriver.java]: We should have these suffix names as constants in some file. HTSJDK has also defined SAM/BAM/CRAM constants., e.g., http://static.javadoc.io/com.github.samtools/htsjdk/1.132/htsjdk/samtools/BamFileIoUtils.html#BAM_FILE_EXTENSION

**Comment by bayolau** [comment]
@yunfeiguo looks good for me. Do you have other cosmetic changes you wanted to make before? now is the time to do it

**Comment by yunfeiguo** [comment]
@bayolau, thanks! Since I enjoyed using Spec for its versatility, I put suffixes and file types into enum. The implementation might not be the a good one, though. For other changes, I want to step back for now, haha...

**Comment by yunfeiguo** [comment]
@bayolau could you please merge? Thanks.

**Pull request merged**